test/vm-upload

```diff
@@ -2,24 +2,40 @@
 
 # The default settings here should match one of the default
 # download stores. These are usually cockpit/images instances
-UPLOAD = "209.132.184.69:8493"
+UPLOAD = "https://209.132.184.69:8493/"
 TOKEN = "~/.config/github-token"
 
 import argparse
+import getpass
 import os
+import socket
 import subprocess
 import sys
-
-from common import testinfra
+import urlparse
 
 BASE = os.path.dirname(__file__)
 IMAGES = os.path.join(BASE, "images")
 DATA = os.path.join(os.environ.get("TEST_DATA", BASE), "images")
 
 def upload(store, source):
     ca = os.path.join(BASE, "common", "ca.pem")
-    (address, delim, port) = store.partition(":")
-    resolve = "cockpit-tests:{1}:{0}".format(address, port)
+    url = urlparse.urlparse(store)
+
+    # Parse out the user name if present
+    netloc = url.netloc
+    if "@" in netloc:
+        (user, delim, netloc) = netloc.partition("@")
+    else:
+        user = getpass.getuser()
+
+    # Parse out the actual address to connect to and override certificate info
+    (address, delim, port) = netloc.partition(":")
+    ai = socket.getaddrinfo(address, int(port), socket.AF_INET, 0, socket.IPPROTO_TCP)
```

Comment:
They aren't part of the actual tuple, but read only attributes. They are in the docs here https://docs.python.org/2/library/urlparse.html